### PR TITLE
Mount cloud-config files for kube-apiserver & kube-controllermanager pods.

### PR DESCRIPTION
**Pull request by cjcullen**
Also relax the MASTER check to be =~ .*-master$
EDIT: Not doing this in this PR anymore.

### CODE REVIEW

**Comment by yujuhong** [comment]
Assigned to @ArtfulCoder he wrote the manifest files.

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ for k,v in yaml.load(sys.stdin).iteritems():
 ''' < "${kube_env_yaml}")
 
   # Infer master status from hostname
-  if [[ $(hostname) == "${INSTANCE_PREFIX}-master" ]]; then
+  if [[ $(hostname) =~ .*-master$ ]]; then
```
Comment (ArtfulCoder) [inline cluster/gce/configure-vm.sh]: this should be a seperate checkin

File: cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest [hunk 2]
```diff
@@ -12,11 +14,15 @@
 {% if grains.cloud == 'gce' -%}
   {% if grains.cloud_config is defined -%}
     {% set cloud_config = "--cloud_config=" + grains.cloud_config -%}
+    {% set cloud_config_mount = "{\"name\": \"cloudconfigmount\",\"mountPath\": \"" + grains.cloud_config + "\", \"readOnly\": true}," -%}
```
Comment (ArtfulCoder) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: the logic for gce and aws is the same.
can you use an 'or' clause ? http://jinja.pocoo.org/docs/dev/templates/#list-of-control-structures

File: cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest [hunk 3]
```diff
@@ -26,11 +28,15 @@
 {% if grains.cloud == 'gce' -%}
   {% if grains.cloud_config is defined -%}
     {% set cloud_config = "--cloud_config=" + grains.cloud_config -%}
+    {% set cloud_config_mount = "{\"name\": \"cloudconfigmount\",\"mountPath\": \"" + grains.cloud_config + "\", \"readOnly\": true}," -%}
```
Comment (ArtfulCoder) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: can we combine the logic for gce and aws using an or clause?

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (cjcullen) [inline cluster/gce/configure-vm.sh]: Done.

File: cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest (hunk 2 above)
Comment (cjcullen) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: I rebased, and it appears somebody else came up with a nice way, so I repeated that here.

File: cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest (hunk 3 above)
Comment (cjcullen) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: Already done for me :)

**Comment by erictune** [comment]
@ArtfulCoder PTAL

**Comment by cjcullen** [comment]
This commit is also included in #9292. This is needed for #3168.

**Comment by ArtfulCoder** [comment]
LGTM

**Comment by cjcullen** [comment]
Going to abandon this PR. These changes will go in with the rest of #9292.

**Pull request closed without merging**